Sources/EdgeNetworkHandlers/EdgeNetworkService.swift

```diff
@@ -71,15 +71,19 @@ class EdgeNetworkService {
     ///   - requestBody: `EdgeRequest` containing the encoded events
     ///   - requestHeaders: request HTTP headers
     ///   - responseCallback: `ResponseCallback` to be invoked once the server response is received
-    ///   - completion: a closure that is invoked with true if the hit should not be retried, false if the hit should be retried
-    func doRequest(url: URL, requestBody: EdgeRequest, requestHeaders: [String: String]? = [:], responseCallback: ResponseCallback, completion: @escaping (Bool) -> Void) {
+    ///   - completion: a closure that is invoked with true if the hit should not be retried, false if the hit should be retried, along with the time interval that should elapse before retrying the hit
+    func doRequest(url: URL,
+                   requestBody: EdgeRequest,
+                   requestHeaders: [String: String]? = [:],
+                   responseCallback: ResponseCallback,
+                   completion: @escaping (Bool, TimeInterval) -> Void) {
```

Comment:
We can use `TimeInterval?`

Reply:
Yes, it's just a double under the hood.

Reply:
Sorry, I misunderstood this context. I read it as "Can we use a `TimeInterval`?", Not can we use an optional time interval.

As stated in another comment, it's not possible since TimeInterval is a Double which cannot take on nil.